library/Booster/Pattern/Util.hs

```diff
@@ -113,11 +117,29 @@ modifyVariablesInT f = cata $ \case
 modifyVarName :: (VarName -> VarName) -> Variable -> Variable
 modifyVarName f v = v{variableName = f v.variableName}
 
+{- | Strip variable provenance prefixes introduced
+in "Syntax.ParsedKore.Internalize.internationalization"
+-}
+stripVarOriginPrefix :: VarName -> VarName
+stripVarOriginPrefix name =
+    let noRule = BS.stripPrefix "Rule#" name
+        noEx = BS.stripPrefix "Ex#" name
+     in fromMaybe name $ noRule <|> noEx
+
 freshenVar :: Variable -> Set Variable -> Variable
 freshenVar v@Variable{variableName = vn} vs
-    | v `Set.member` vs = freshenVar v{variableName = vn <> "'"} vs
+    | v `Set.member` vs = freshenVar v{variableName = incrementNameCounter vn} vs
     | otherwise = v
 
+incrementNameCounter :: VarName -> VarName
+incrementNameCounter vname =
+    let (name, counter) = BS.spanEnd A.isDigit_w8 vname
+        parsedCounter = A.parseOnly @Int A.decimal counter
+        newCounter = case parsedCounter of
+            Right ok -> ok + 1
+            Left _err -> 0
+     in name <> (BS.pack . map (fromIntegral . ord) . show) newCounter
```

Comment:
I dont understand what `map (fromIntegral . ord) . show` is meant to be doing. can you add a comment/simplify this code?

Reply:
It unparses the `Int` variable counter back into a sequence of bytes representing the digits. Don't know an easy way to simplify that, so I extracted the expression into a variable and added a comment.